tests/qa/test_test_config.py

```diff
@@ -29,3 +37,53 @@ def test_test_config_access_om2():
         print(f"Test stdout: {result.stdout}\nTest stderr: {result.stderr}")
 
     assert result.returncode == 0
+
+
+@pytest.mark.parametrize(
+    "repo_name, model_version, test_content",
+    [
+        # Release with spack.location file
+        ("ACCESS-OM2", "2024.03.0", "access-om2-2024_03_0"),
+        # Release with Gadi.spack.location file
+        ("ACCESS-OM3", "2025.01.1", "access-om3-2025_01_1"),
+    ],
+)
+def test_get_spack_location_file(repo_name, model_version, test_content):
+    """
+    Test to check that get_spack_location_file runs without error
+    for a couple ACCESS-NRI releases
+    """
+    spack_location = get_spack_location_file(repo_name, model_version)
+    assert test_content in spack_location
+
+
+def test_get_spack_location_file_no_release_artefact():
+    """
+    Test that an error is raised when the release artefact is not found
+    """
+    with pytest.raises(AssertionError, match=r"Failed to find release .*"):
+        get_spack_location_file("fake-repo-name", "fake.module.version")
+
+
+def mock_request_get(url, *args, **kwargs):
+    """Custom side effect function for mocking requests.get to pass the initial
+    initial request (e.g that a release artefact exists)
+    but returns a 404 status for subsequent requests.get calls
+    """
+    response = Mock()
+    if url == "https://github.com/ACCESS-NRI/fake-repo/releases/tag/fake-version":
+        response.status_code = 200
+    else:
+        response.status_code = 404
```

Comment:
```suggestion
    if url == "https://github.com/ACCESS-NRI/fake-repo/releases/tag/fake-version":
        response.status_code = 200
    else:
        response.status_code = 404
```
Why do you need this differentiation within this test? 
Are you planning to use `mock_request_get` in other tests?

Otherwise I would simply set `response.status_code = 404` (or any number but `200`)

Reply:
Oh I see now you're using `requests` also above.

Reply:
Yeah, I use requests to check if a release artefact exists first before trying to download a file